src/metatrain/utils/additive/composition.py

```diff
@@ -373,97 +117,47 @@ def restart(self, dataset_info: DatasetInfo) -> "CompositionModel":
         self.dataset_info = merged_info
 
         # register new outputs
-        for target_name, target in self.new_targets.items():
-            self._add_output(target_name, target)
+        for target_name, target_info in self.new_targets.items():
+            self._add_output(target_name, target_info)
 
         return self
 
     def forward(
         self,
         systems: List[System],
         outputs: Dict[str, ModelOutput],
-        selected_atoms: Optional[Labels] = None,
+        selected_samples: Optional[Labels] = None,
```

Comment:
this should still be `selected_atoms`, even in the `sample_kind="system"` one would want to only include a subset of the atoms in the calculation.

Reply:
I was trying to pre-empt a breaking change once pairs are available. I think samples are more general and more correct - in most other places in our stack we do "sample selection" and not "atom selection". In metatrain I think this shoudl be changed for all models, but we can start with this one. Thoughts?